internal/wasm/jit/engine.go

```diff
@@ -669,11 +661,15 @@ func (e *engine) addCompiledFunction(addr wasm.FunctionAddress, compiled *compil
 }
 
 func compileHostFunction(f *wasm.FunctionInstance) (*compiledFunction, error) {
-	compiler, err := newCompiler(f, nil)
+	compiler, done, err := newCompiler(f, nil)
 	if err != nil {
 		return nil, err
 	}
```

Comment:
This error case will cause the dead lock of the mutex. Maybe better to call `done` before this block, as long as it's not nil

Reply:
nice!